### ISSUE TITLE
Model is not learning problem

### ISSUE THREAD

**Issue by DRMALEK** [wasiahmad/NeuralCodeSum#27]
Hi,

I'm trying to train the Transformer model on a custom dataset, however, the model fails to learn and just output the same prediction for all the inputs. More than that, due to the early stopping mechanism, it always stops after the 20 epoch and not continue forward. 

I tried to change the learning rate, the optimizer, and then stopping, however, there is no improvement.

Thanks.

**Comment by wasiahmad** [comment]
You should post this question to some forum. We are unable to help you to fix issues if you are experimenting with a custom dataset because we have no knowledge about that. Closing this issue.

**Comment by DRMALEK** [comment]
Actually, I prepared the custom dataset, the same way the dataset that used in your paper, the same tokenization techniques, and the same ratios. But instead of java projects, I used Android projects instead. 

Thanks.

**Comment by wasiahmad** [comment]
How many training examples you have?
